### ISSUE TITLE
Problem with FVDI+NVDI vr_transfer_bits()

### ISSUE THREAD

**Issue by AtariTrek** [vinriviere/fvdi#45]
"fvdi + nvdi" works to some degree
vr_transfer_bits() seems incomplete
fvdi +nvdi crash at a null address to the screen with vr_transfer_bits()

And vs_ctab_entry() commands seem not fully supported, which makes fvdi uncomplete compared to nvdi.

Thank you.

**Comment by mikrosk** [comment]
@AtariTrek can you provide more details? I.e. how do you run fvdi and nvdi (auto, mint.cnf, what order, what machine) and some code snippet to test?

**Comment by AtariTrek** [comment]
Hi Miro!
The program imgview (an excellent program) crash when "fulling" the screen
image while using fvdi.
While using plain nvdi it doesn't.
https://atari.gfabasic.net/
This is due to the fact that a NULL address with vr_transfer_bits() makes
fvdi crash, while the nvdi docs clearly states that a Null address can be
used.
I hope this helps ;)
Thank you

Den ons 18 okt. 2023 kl 20:25 skrev Miro Kropáček ***@***.***

**Comment by mikrosk** [comment]
Ah ok, so it's "just" a plain fVDI bug. I was confused with the combination with NVDI.

**Comment by AtariTrek** [comment]
Well both yes and no.
It is with the combination fvdi+nvdi that makes vr_transfer_bits() crash
together with dither_bits
thank you

Den tis 24 okt. 2023 kl 11:11 skrev Miro Kropáček ***@***.***

**Comment by mikrosk** [comment]
Can you please test just fvdi alone then?

**Comment by AtariTrek** [comment]
Dither_bits does not exist with fvdi alone, it is needed with nvdi.
If you install the program I provided (imgview+modules) you will be able to
replicate the problem.
The other modes perhaps does the same thing with vr_transfer_bits(), I am
not sure.
I use aranym windows and do not have fvdi installed at the moment.
Just nvdi and blowup.

Den tis 24 okt. 2023 kl 15:07 skrev Miro Kropáček ***@***.***

**Comment by AtariTrek** [comment]
I have now installed fvdi, but it shouldn't matter for test purposes since
I can not test vr_transfer_bits(/) without dithering and nvdi 5.03?

Den tis 24 okt. 2023 kl 20:36 skrev Anders Lindström ***@***.***>:

**Comment by AtariTrek** [comment]
The problem would most surely be fvdi since the null adress works with
nvdi only without problems.

Den ons 25 okt. 2023 kl 07:43 skrev Anders Lindström ***@***.***>:

**Comment by lpgb** [comment]
To clarify, boot fvdi + nvdi, in my case aranym was used in 256 color mode as test case. Start ImgView and load an image then press ESCape. It does a full screen non-gem presentation of the image. I was able to reproduce the crash. See attached image from the NVDI docs. If I had to guess fvdi isn't handling the NULL shortcut correctly.

<img width="687" height="460" alt="Image" src="https://github.com/user-attachments/assets/c931ae72-08b3-42c3-a6f4-038ddbff75fd" />

**Comment by mikrosk** [comment]
I have briefly checked fvdi sources, I can't see an obvious bug in src_dm / dst_dm handling. How is NVDI and FVDI started? (auto order, mint.cnf, I guess it is 5.03?...)

**Comment by lpgb** [comment]
> I have briefly checked fvdi sources, I can't see an obvious bug in src_dm / dst_dm handling. How is NVDI and FVDI started? (auto order, mint.cnf, I guess it is 5.03?...)

Yes, 5.03. It's all preinstalled, used this: https://sites.google.com/site/beebox68k/easyaramint
